src/mdns.rs

```diff
@@ -0,0 +1,444 @@
+use esp_idf_sys::*;
+
+extern crate alloc;
+use alloc::vec::Vec;
+
+use std::ffi::CStr;
+use std::ffi::CString;
+
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
```

Comment:
This needs to use the types exposed from the `ipv4` module instead. This is a module of `embedded_svc`.

Reply:
Which IpAddr and Ipv6Addrshould We use?
Does embedded_svc plan to export that?

Reply:
You are right, embedded-svc does not yet expose these. Keep them as they are, ill fix post-commit

Reply:
We fixed it.